File: euclidean-algorithm/EuclideanAlgorithm.java
public class EuclideanAlgorithm {
    public static int gcd(int a, int b) {
        while (b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
    
    public static int gcdRecursive(int a, int b) {
        return b == 0 ? a : gcdRecursive(b, a % b);
    }
    
    public static int lcm(int a, int b) {
        return (a * b) / gcd(a, b);
    }
    
    public static int[] extendedGcd(int a, int b) {
        if (b == 0) {
            return new int[]{a, 1, 0};
        }
        int[] result = extendedGcd(b, a % b);
        int gcd = result[0];
        int x1 = result[1];
        int y1 = result[2];
        
        int x = y1;
        int y = x1 - (a / b) * y1;
        
        return new int[]{gcd, x, y};
    }
    
    public static void main(String[] args) {
        System.out.println("GCD: " + gcd(48, 18));
        System.out.println("GCD recursive: " + gcdRecursive(48, 18));
        System.out.println("LCM: " + lcm(48, 18));
        
        int[] result = extendedGcd(48, 18);
        System.out.printf("Extended GCD: gcd=%d, x=%d, y=%d%n", result[0], result[1], result[2]);
    }
}

File: euclidean-algorithm/euclidean_algorithm.rs
fn gcd(mut a: i32, mut b: i32) -> i32 {
    while b != 0 {
        let temp = b;
        b = a % b;
        a = temp;
    }
    a
}

fn gcd_recursive(a: i32, b: i32) -> i32 {
    if b == 0 { a } else { gcd_recursive(b, a % b) }
}

fn lcm(a: i32, b: i32) -> i32 {
    (a * b) / gcd(a, b)
}

fn extended_gcd(a: i32, b: i32) -> (i32, i32, i32) {
    if b == 0 {
        (a, 1, 0)
    } else {
        let (gcd, x1, y1) = extended_gcd(b, a % b);
        let x = y1;
        let y = x1 - (a / b) * y1;
        (gcd, x, y)
    }
}

fn main() {
    println!("GCD: {}", gcd(48, 18));
    println!("GCD recursive: {}", gcd_recursive(48, 18));
    println!("LCM: {}", lcm(48, 18));
    println!("Extended GCD: {:?}", extended_gcd(48, 18));
}